src/main/java/edu/ucla/library/avpairtree/verticles/WaveformVerticle.java

```diff
@@ -55,13 +65,13 @@ public void start(final Promise<Void> aPromise) {
         try {
             final Process which = new ProcessBuilder(cmd).start();
             final int exitValue = which.waitFor();
-            final InputStream stdout;
             final String cmdResult;
 
-            stdout = which.getInputStream();
-            cmdResult =
-                    LOGGER.getMessage(MessageCodes.AVPT_015, cmdline, exitValue, new String(stdout.readAllBytes()));
-            stdout.close();
+            try (InputStream stdIn = which.getInputStream()) {
```

Comment:
`which.getInputStream()` does connect to the subprocess "stdout", but is connected to "stdin"? Either way, I do think the name `stdout` is ambiguous and should be renamed.

Reply:
I'm fine with renaming to something less ambiguous.

Reply:
`subprocessOutput`?

Reply:
Isn't that what we're reading bytes from, though -- the subprocess output? (via an InputStream that's connected to the subprocess' stdout, of course). I suppose it's something that could go either way -- it's represents both an output (of the subprocess) and an input (to the calling code)

Reply:
The name `inStream` is, to me, redundant, because I can see that it's an InputStream by its type

Reply:
To me it's like exception and parent exception. You may have one exception that wraps another exception, but the exception you're dealing with is the one in hand, not the parent. You have an output stream from the which process, but what you have here is the input stream that's reading from that output stream.

Another way to look at it: `output = which.getInputStream()` ... you're getting an input stream the output -> input is handled within the which. You're reading the results of which here.

Reply:
> You have an output stream from the which process, but what you have here is the input stream that's reading from that output stream.

Doesn't the variable represent both?

> what you have here is the input stream that's reading from that output stream
> you're getting an input stream
> You're _reading_ the results of which here.

Why doesn't the type `InputStream` sufficiently convey this?

Reply:
1. No, I don't think it does. The output process is subsumed by the reading of it.

2. I'd same for the same reason `ARK ark` is not redundant. One is a type and one is a name. It makes sense, to me, to name the variable according to the type. Otherwise it would be something like `ARK string` because it's a string that's been made into an ARK.

Reply:
The compromise would be `InputStream outputInputStream = which.getInputStream()` but to me that seems unnecessary.

Reply:
I suppose the reason why I think the variable does represent both is because I just think of it as a source of data -- whether it's an `InputStream` or a byte array or something else, it contains the output of the subprocess, which in this case happens to be obtained by calling `readAllBytes`. While it having the type `InputStream` defines exactly how the data is obtained, at a conceptual level all we're doing is getting the output from the subprocess.

I think I'd also disagree re: `ARK ark` not being redundant. That is also redundant in my view -- I'd prefer something like `ARK identifier` or `ARK collectionIdentifier`. Although, we usually represent them as `String` anyway.

I don't think this is a very productive discussion since I think we're getting into philosophy here -- I read [this comment](https://github.com/UCLALibrary/av-pairtree/pull/13#discussion_r714229232) as you wanting a suggestion from me, but I can also accept your choice of name.

Reply:
Yeah, I was asking for a suggestion, but I think we hit the philosophical difference there. I wasn't expecting a suggestion from the underlying stream. If you can accept the choice of name, I'd agree... let's move on.

Reply:
But to continue the philosophical discussion<g>, I don't see `ark` and `identifier` as different things but as levels of specificity for the same thing, while and input and output stream are different things.

Reply:
Actually, I think I'll change my choice of name to inStream because it's not really stdIn... that was a bad choice.